File: frontend/src/types/class.ts
/**
 * @deprecated 请使用 @/types/organization 中的 SchoolClass 类型
 * V1 类型保留用于兼容旧组件，新开发请使用 V2 类型
 *
 * 主要变更:
 * - Class -> SchoolClass
 * - departmentId -> orgUnitId
 * - status: number -> status: ClassStatus (字符串枚举)
 * - teacherId/teacherName -> teacherAssignments 数组
 *
 * 注意: ClassDormitoryInfo 暂无 V2 替代，继续使用本文件
 */

// 班级相关类型定义

export interface Class {
  id: number | string
  className: string
  classCode: string
  gradeLevel: number
  gradeId?: number | string
  orgUnitId: number | string
  orgUnitName: string
  majorId?: number | string
  majorName?: string
  majorDirectionId?: number | string
  classSequence?: number
  educationSystem?: string
  skillLevel?: string
  duration?: number
  grade: string
  teacherId?: number | string
  teacherName?: string
  assistantTeacherId?: number | string
  assistantTeacherName?: string
  studentCount: number
  maxStudents?: number
  classroomLocation?: string
  enrollmentYear?: number
  graduationYear?: number
  classType?: number
  classTypeName?: string
  status: number
  statusName?: string
  createdAt: string
  updatedAt?: string
}

export interface ClassQueryParams {
  className?: string
  orgUnitId?: number | string | null
  majorId?: number | string | null
  teacherName?: string
  status?: number | null
  pageNum?: number
  pageSize?: number
}

export interface ClassFormData {
  className: string
  classCode: string
  gradeLevel: number | null
  orgUnitId: number | string | null
  majorId?: number | string | null
  teacherId?: number | string | null
  assistantTeacherId?: number | string | null
  classroomLocation?: string
  enrollmentYear: number | null
  graduationYear: number | null
  classType: number
  status: number
}

/**
 * 班级宿舍关联信息
 */
export interface ClassDormitoryInfo {
  id: number | string
  dormitoryId: number | string
  dormitoryNo: string  // 房间号
  buildingNo?: string  // 楼号
  buildingName?: string  // 楼宇名称
  floorNumber?: number  // 楼层
  bedCapacity?: number  // 床位容量规格
  bedCount?: number  // 实际床位数
  roomUsageType?: number  // 房间用途类型
  occupiedBeds?: number  // 已占用床位
  allocatedBeds?: number  // 分配给班级的床位数
  genderType?: number  // 性别类型
  status?: number
}

File: backend/src/main/java/com/school/management/entity/Class.java
package com.school.management.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.Year;

/**
 * 班级实体
 *
 * @author system
 * @since 1.0.0
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("classes")
public class Class extends BaseEntity {

    /**
     * 班级名称
     */
    private String className;

    /**
     * 班级编码
     */
    private String classCode;

    /**
     * 年级
     */
    private Integer gradeLevel;

    /**
     * 所属组织单元ID
     */
    private Long orgUnitId;

    /**
     * 年级ID
     */
    private Long gradeId;

    /**
     * 专业ID
     */
    private Long majorId;

    /**
     * 专业方向ID
     */
    private Long majorDirectionId;

    /**
     * 班主任ID
     */
    private Long teacherId;

    /**
     * 副班主任ID
     */
    private Long assistantTeacherId;

    /**
     * 学生数量
     */
    private Integer studentCount;

    /**
     * 教室位置
     */
    private String classroomLocation;

    /**
     * 入学年份
     */
    private Integer enrollmentYear;

    /**
     * 毕业年份
     */
    private Integer graduationYear;

    /**
     * 年级(格式化字段,非数据库字段)
     */
    @TableField(exist = false)
    private Integer grade;

    /**
     * 班级类型: 1普通班 2重点班 3实验班
     */
    private Integer classType;

    /**
     * 状态: 1启用 0禁用
     */
    private Integer status;

    // ========== 关联字段（非数据库字段） ==========

    /**
     * 组织单元名称(关联查询)
     */
    @TableField(exist = false)
    private String orgUnitName;

    /**
     * 组织单元信息(关联查询)
     */
    @TableField(exist = false)
    private Department orgUnit;

    /**
     * 专业信息(关联查询)
     */
    @TableField(exist = false)
    private Major major;

    /**
     * 专业名称(关联查询)
     */
    @TableField(exist = false)
    private String majorName;

    /**
     * 班主任姓名(关联查询)
     */
    @TableField(exist = false)
    private String teacherName;

    /**
     * 班主任信息(关联查询)
     */
    @TableField(exist = false)
    private User teacher;

    /**
     * 副班主任姓名(关联查询)
     */
    @TableField(exist = false)
    private String assistantTeacherName;

    /**
     * 副班主任信息(关联查询)
     */
    @TableField(exist = false)
    private User assistantTeacher;
}